geopandas/io/sql.py

```diff
@@ -96,3 +96,222 @@ def load_geom_text(x):
                 crs = {"init": "epsg:{}".format(srid)}
 
     return GeoDataFrame(df, crs=crs, geometry=geom_col)
+
+
+def get_geometry_type(gdf):
+    """
+    Get basic geometry type of a GeoDataFrame,
+    and information if the gdf contains Geometry Collections."""
+    geom_types = list(gdf.geometry.geom_type.unique())
+    geom_collection = False
+
+    # Get the basic geometry type
+    basic_types = []
+    for gt in geom_types:
+        if "Multi" in gt:
+            geom_collection = True
+            basic_types.append(gt.replace("Multi", ""))
+        else:
+            basic_types.append(gt)
+    geom_types = list(set(basic_types))
+
+    # Check for mixed geometry types
+    assert len(geom_types) < 2, "GeoDataFrame contains mixed geometry types."
```

Comment:
Can you make a check and raise specific error instead of AssertionError to be more explicit?

```
if len(geom_types) < 2:
    raise...
```

Reply:
Actually this "error" should not be an error at all but this should return GeometryCollection type. After diving more into the GeoAlchemy docs found that this indeed is possible..

Reply:
If there is a mix of Geometry-types, they will be now inserted with PostGIS type `GEOMETRY`.